programs/mango-v4/src/health/account_retriever.rs

```diff
@@ -134,7 +134,14 @@ impl<T: KeyedAccountReader> AccountRetriever for FixedOrderAccountRetriever<T> {
             })?;
 
         let oracle_index = self.n_banks + active_token_position_index;
-        let oracle_price = self.oracle_price_bank(oracle_index, bank).with_context(|| {
+        let oracle = &self.ais[oracle_index];
+        let fallback_opt = self
+            .fallback_oracle_ais
+            .iter()
+            .find(|ai| ai.key() == &bank.fallback_oracle);
```

Comment:
I guess scanning through the list of fallback oracles every time is cheap enough as long as there's only a few...

Reply:
Maybe making a HashMap like the ScanningAccountRetirever does would be preferable? Only one scan is guaranteed in that case:
https://github.com/blockworks-foundation/mango-v4/pull/790/commits/91cd42f0443b1f6187aecf67d47d70b7b1a51ad5#diff-0b7a655c1f66e408b1fd89404a260921036deb705eaef311f4ccce13974edd93R373

Reply:
as far as CU goes, even with 8 tokens it's only max 100 difference (based on `test_health_compute_tokens_fallback_oracles`)

Reply:
Thanks for checking, yeah keep scanning then

Reply:
We could also save CU by doing the scan only if the fallback is actually needed (but since CU cost looks ok, this optimization isn't required imo)